spmd/tensor/parallel/style.py

```diff
@@ -28,14 +28,34 @@ class ParallelStyle(ABC):
     _prepare_output: Optional[_Prepare_Output_Func_Type]
 
 
+class RowwiseParallel(ParallelStyle):
+    """
+    Partitioning the row of a module.
+    We assume the input to be a sharded :class:``DTensor`` and output to be a replicated :class:``DTensor``.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(make_input_shard_1d, make_output_replicate_1d)
+
+
+class ColwiseParallel(ParallelStyle):
+    """
+    Partitioning the column of a tensor or module.
+    We assume the input to be a replicated :class:``DTensor`` and output to be a sharded :class:``DTensor``.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(make_input_replicate_1d, None)
```

Comment:
As per discussion with Wanchao, we believe we can use make_output_replicate_1d for prepare_out as well. So we removed Optional for both prepare_input and prepare_output.